tuning/utils/preprocessing_utils.py

```diff
@@ -0,0 +1,144 @@
+            return DataCollatorForSeq2Seq(tokenizer=tokenizer, padding=True, max_length=max_sequence_length)
+        if dataset_text_field is None and response_template is not None:
+            raise ValueError("Packing is disabled, but no dataset_text_field is provided")
+        if response_template is None and dataset_text_field is not None:
+            raise ValueError("Packing is disabled, but no response template is provided")
+        # TODO: near term - how response template ids are parsed out needs to be cleaned.
+        # The [2:] here applies if response template has \n prefix, it is needed to strip \n,
+        # otherwise template is not found. We will create issue to clean this out after we discuss
+        # data formats and collators we will support.
+        response_template_ids = tokenizer.encode(
+            response_template, add_special_tokens=False
+        )[2:]
+        return DataCollatorForCompletionOnlyLM(
+            response_template=response_template_ids,
+            tokenizer=tokenizer,
+            ignore_index=configs.IGNORE_INDEX
+        )
+
+
+def get_formatted_dataset(data_path: str, dataset_text_field, tokenizer):
+    format_dataset = lambda example: {  # pylint: disable=unnecessary-lambda-assignment
+        f"{dataset_text_field}": example[f"{dataset_text_field}"]
+        + tokenizer.eos_token
+    }
+    json_dataset = datasets.load_dataset("json", data_files=data_path)
+    return json_dataset.map(format_dataset)['train'] # HACK - just do both datasets separately
+
+def get_preprocessed_dataset(data_path: str, tokenizer: AutoTokenizer, max_sequence_length: int, input_field_name: str, output_field_name: str):
+    dataset = load_hf_dataset_from_jsonl_file(data_path, input_field_name, output_field_name)
+    return dataset.map(
+        preprocess_and_tokenize,
+        fn_kwargs={
+            "tokenizer": tokenizer,
+            "max_seq_length": max_sequence_length,
+            "input_field_name": input_field_name,
+            "output_field_name": output_field_name,            
+        },
+        remove_columns=[input_field_name, output_field_name],
+    )
+
+### Utils for loading the data from disk in supported formats [currently only jsonl]
+def load_hf_dataset_from_jsonl_file(data_path, input_field_name: str, output_field_name: str):
+    if input_field_name == output_field_name:
+        raise ValueError("Input field name and output field name should not match!")
+    def get_jsonl_object():
+        jsonl_file = open(data_path, "r")
+        data_stream = [json.loads(line) for line in jsonl_file]
+        for data in data_stream:
+            yield {
+                input_field_name: data[input_field_name],
+                output_field_name: data[output_field_name]
+            }
+    return Dataset.from_generator(
+        get_jsonl_object
+    )
+
+### Utils for custom masking / manipulating input / output strs, etc
+def combine_sequence(input_element: str, output_element: str):
```

Comment:
with my upcoming change to accept a template in API - which will be like verbalizer field. In future we ll need to apply template while combining sequence - which will be a minor addition, that way we can accept "input/output" + custom template. Need not worry about that now though

Reply:
Cool sounds good - that is partially why I wrote things this way, even though the input/output are hardcoded in the thing calling this, so we can just pass everything through here as needed

Reply:
are the templates going to be jinja style like this? https://huggingface.co/docs/transformers/en/chat_templating